src/creatures/players/cast/cast_viewer.hpp

```diff
@@ -0,0 +1,261 @@
+	void removeCaster();
+	bool canSee(const Position &pos) const;
+	uint32_t getIP() const;
+	void sendStats();
+	void sendPing();
+	void logout(bool displayEffect, bool forceLogout);
+	void sendAddContainerItem(uint8_t cid, uint16_t slot, std::shared_ptr<Item> item);
+	void sendUpdateContainerItem(uint8_t cid, uint16_t slot, std::shared_ptr<Item> item);
+	void sendRemoveContainerItem(uint8_t cid, uint16_t slot, std::shared_ptr<Item> lastItem);
+	void sendUpdatedVIPStatus(uint32_t guid, VipStatus_t newStatus);
+	void sendVIP(uint32_t guid, const std::string &name, const std::string &description, uint32_t icon, bool notify, VipStatus_t status);
+	void sendVIPGroups();
+	void sendClosePrivate(uint16_t channelId);
+	void sendFYIBox(const std::string &message);
+	uint32_t getVersion() const;
+	void disconnect();
+	void sendCreatureSkull(std::shared_ptr<Creature> creature) const;
+	void sendAddTileItem(const Position &pos, uint32_t stackpos, std::shared_ptr<Item> item);
+	void sendUpdateTileItem(const Position &pos, uint32_t stackpos, std::shared_ptr<Item> item);
+	void sendRemoveTileThing(const Position &pos, int32_t stackpos);
+	void sendUpdateTile(std::shared_ptr<Tile> tile, const Position &pos);
+	void sendChannelMessage(const std::string &author, const std::string &message, SpeakClasses type, uint16_t channel);
+	void sendMoveCreature(std::shared_ptr<Creature> creature, const Position &newPos, int32_t newStackPos, const Position &oldPos, int32_t oldStackPos, bool teleport);
+	void sendCreatureTurn(std::shared_ptr<Creature> creature, int32_t stackpos);
+	void sendForgeResult(ForgeAction_t actionType, uint16_t leftItemId, uint8_t leftTier, uint16_t rightItemId, uint8_t rightTier, bool success, uint8_t bonus, uint8_t coreCount, bool convergence) const;
+	void sendPrivateMessage(std::shared_ptr<Player> speaker, SpeakClasses type, const std::string &text);
+	void sendCreatureSquare(std::shared_ptr<Creature> creature, SquareColor_t color);
+	void sendCreatureOutfit(std::shared_ptr<Creature> creature, const Outfit_t &outfit);
+	void sendCreatureLight(std::shared_ptr<Creature> creature);
+	void sendCreatureWalkthrough(std::shared_ptr<Creature> creature, bool walkthrough);
+	void sendCreatureShield(std::shared_ptr<Creature> creature);
+	void sendContainer(uint8_t cid, std::shared_ptr<Container> container, bool hasParent, uint16_t firstIndex);
+	void sendInventoryItem(Slots_t slot, std::shared_ptr<Item> item);
+	void sendCancelMessage(const std::string &msg) const;
+	void sendCancelTarget() const;
+	void sendCancelWalk() const;
+	void sendChangeSpeed(std::shared_ptr<Creature> creature, uint32_t newSpeed) const;
+	void sendCreatureHealth(std::shared_ptr<Creature> creature) const;
+	void sendDistanceShoot(const Position &from, const Position &to, unsigned char type) const;
+	void sendCreatePrivateChannel(uint16_t channelId, const std::string &channelName);
+	void sendIcons(uint32_t icons) const;
+	void sendMagicEffect(const Position &pos, uint8_t type) const;
```

Comment:
It should be 16 bits.

```suggestion
	void sendMagicEffect(const Position &pos, uint16_t type) const;
```